public/app/features/alerting/unified/components/silences/SilencedInstancesPreview.tsx

```diff
@@ -14,25 +18,43 @@ import { AmAlertStateTag } from './AmAlertStateTag';
 
 interface Props {
   amSourceName: string;
-  matchers: Matcher[];
+  matchers: MatcherFieldValue[];
+  ruleUid?: string;
 }
 
-export const SilencedInstancesPreview = ({ amSourceName, matchers }: Props) => {
-  const { useGetAlertmanagerAlertsQuery } = alertmanagerApi;
+const useDebouncedDeepCompare = (cb: () => void, debounceMs: number, dependencies: unknown[]) => {
+  const [state, setState] = useState<unknown[]>();
+
+  useDebounce(cb, debounceMs, [state]);
+
+  useDeepCompareEffect(() => {
+    setState(dependencies);
+  }, [dependencies]);
+};
+
+export const SilencedInstancesPreview = ({ amSourceName, matchers: inputMatchers, ruleUid }: Props) => {
+  const matchers: Matcher[] = [
+    ...(ruleUid ? [{ name: MATCHER_ALERT_RULE_UID, value: ruleUid, operator: MatcherOperator.equal }] : []),
+    ...inputMatchers,
+  ].map(matcherFieldToMatcher);
+  const useLazyQuery = alertmanagerApi.endpoints.getAlertmanagerAlerts.useLazyQuery;
   const styles = useStyles2(getStyles);
   const columns = useColumns();
 
   // By default the form contains an empty matcher - with empty name and value and = operator
   // We don't want to fetch previews for empty matchers as it results in all alerts returned
-  const hasValidMatchers = matchers.some((matcher) => matcher.value && matcher.name);
-
-  const {
-    currentData: alerts = [],
-    isFetching,
-    isError,
-  } = useGetAlertmanagerAlertsQuery(
-    { amSourceName, filter: { matchers } },
-    { skip: !hasValidMatchers, refetchOnMountOrArgChange: true }
+  const hasValidMatchers = ruleUid || inputMatchers.some((matcher) => matcher.value && matcher.name);
+
+  const [getAlertmanagerAlerts, { currentData: alerts = [], isFetching, isError }] = useLazyQuery();
+
+  useDebouncedDeepCompare(
+    () => {
+      if (hasValidMatchers || ruleUid) {
```

Comment:
Isnt' `hasValidMatchers` enough?

Reply:
Ah yes - because I added `ruleUid` to the `hasValidMatchers` definition further up. That one _is_ needed, because the form defaults to a rule UID but no additional matchers